docs/sandboxing/README.md

```diff
@@ -10,9 +10,95 @@ There is no clear solution to this yet, but it should take heavy inspiration fro
+- Read-only bind mounts
+  - `/etc/drirc`
+  - `/etc/gai.conf`
+  - `/etc/gnutls`
+  - `/etc/hostname`
+  - `/etc/hosts`
+  - `/etc/je_malloc.conf`
+  - `/etc/localtime`
+  - `/etc/machine-id` - TODO: what are consequences of omitting this
+  - `/etc/os-release` - TODO: what are consequences of omitting this
+  - `/etc/resolv.conf`
+  - `/etc/selinux`
+  - `/etc/timezone`
+  - `/usr`
+  - `/bin`
+  - `/sbin`
+  - `/lib`
+  - `/lib32`
+  - `/lib64`
+  - `/sys/class`
+  - `/sys/dev/char`
+  - `/sys/devices/pci????:??` - Possibly multiple?
+  - `/sys/devices/system/cpu`
+  - Path to game assets and libraries
+- Read-write bind mounts
+  - Path to game root (`.minecraft`)
+
+##### Desktop integration
+
+- Read-only bind mounts
+  - `$XDG_RUNTIME_DIR/pulse` - Access to PulseAudio daemon
+  - `$XDG_RUNTIME_DIR/pipewire-?` - Access to PipeWire daemon (`pipewire-0` if unspecified)
+  - `$XAUTHORITY` - X11 access
+  - `/tmp/.X11-unix/X?` - X11 access (`X0` if `DESKTOP=:0`)
+  - `$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY` - Wayland access (`$WAYLAND_DISPLAY` can also be an absolute path)
+
+##### Distro-specific configuration
+
+To solve issues with distro-specific symlinks of files, mostly in `/etc` (i.e. `resolv.conf -> /run/systemd/resolve/stub-resolv.conf`), bind mounts SHOULD use canonical file paths for the host side, while preserving the original path in the sandbox.
+
+- Read-only bind mounts
+  - `/nix/store` - Required for Nix installations
+  - `/nix/var/nix/profiles` - Required for Nix installations
+  - `/run/current-system/sw` - Required for NixOS installations
+
+##### Supporting `xdg-open`
+
+By default xdg-open would launch applications inside our sandbox. We can utilize `xdg-desktop-portal` to open links/paths on the host system.
+To do this we *need* dbus, but we neither ensure that it's bound inside the sandboxing, neither do we pass the `DBUS_SESSION_BUS_ADDRESS` variable to the sandbox.
+We don't want to allow full access to dbus, as that could allow sandbox escapes (some endpoints allow executing arbitrary commands on the host).
+The Flatpak project maintains a tool called `xdg-dbus-proxy` that acts as a filtering reverse proxy for dbus.
+
+As we only need access to `org.freedesktop.portal.*` the call to xdg-dbus-proxy is quite simple:
+
+```console
+# PROXIED_BUS_PATH can be an arbitrary path on the host
+$ xdg-dbus-proxy $DBUS_SESSION_BUS_ADDRESS $PROXIED_BUS_PATH --filter '--call=org.freedesktop.portal.*=*' '--broadcast=org.freedesktop.portal.*=@/org/freedesktop/portal/*'
+```
+
+To give the sandbox access to this filtered dbus, the `bwrap` call must be extended by the following two parameters:
+- `--bind $PROXIED_BUS_PATH /tmp/bus` - TODO: can this be read-only?
+- `--setenv DBUS_SESSION_BUS_ADDRESS "unix:path=/tmp/bus"`
+
+This is NOT enough to make xdg-open use portals yet. At its current state, we can convince xdg-open that it should use Portals by pretending to be a Flatpak.
```

Comment:
As per my other comment, flatpak does use a custom `xdg-open` implementation: https://github.com/flatpak/flatpak-xdg-utils
This will use the portal without such environment variable hacks, which should be avoided.

Reply:
They seem to use sockets like my original idea, I'm willing to work on this.

Reply:
Would we expect launchers to ship this binary?

Reply:
Hm... maybe shipping it inside the sandbox binary could work?